### PR TITLE
Introducing netstandard13aot target group in System.IO

### PR DESCRIPTION
**Pull request by tarekgh**
We need to build System.IO that work with corert which we need to get rid of WinRT
dependencies and that will require to have different code in System.IO which require
creating a new target group. This change should be no-op for other target groups.

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard  @ericstj could you please have a look?

File: src/System.IO/src/netstandard15aot/project.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "frameworks": {
+    "netstandard1.5": {
```
Comment (weshaggard) [inline src/System.IO/src/netstandard15aot/project.json]: I thought you were going to make this netstandard13aot.

Comment (jasonwilliams200OK) [inline src/System.IO/src/netstandard15aot/project.json]: Since this has been changed to `netstandard13aot`, is the commit message still valid?
The PR title and commit message: "Introducing `netstandard51aot` target.."

File: dir.props [hunk 2]
```diff
@@ -308,6 +309,13 @@
         <NuGetTargetMoniker>.NETCore,Version=v5.0</NuGetTargetMoniker>
       </PropertyGroup>
     </When>
+    <When Condition="'$(TargetGroup)'=='netstandard13aot'">
+      <PropertyGroup>
+        <PackageTargetFramework>netstandard13aot</PackageTargetFramework>
```
Comment (ericstj) [inline dir.props]: should be just nestandard1.3

File: src/System.IO/src/netstandard15aot/project.json (hunk 1 above)
Comment (ericstj) [inline src/System.IO/src/netstandard15aot/project.json]: Don't do that.  Appending aot to the framework moniker is not valid.  'netstandard13aot' is just an arbitrary string the build tooling is using to enable cross-compilation.

Comment (weshaggard) [inline src/System.IO/src/netstandard15aot/project.json]: Agreed. The project.json should just be netstandard1.3 for the framework. netstandard13aot is just shorthand for our build system.

**Comment by tarekgh** [comment]
@weshaggard & @ericstj I have fixed the code according to your comment and also included the packaging change as Eric described to me.

File: src/System.IO/src/System.IO.csproj [hunk 3]
```diff
@@ -16,16 +16,19 @@
...
     <ResourcesSourceOutputDirectory>None</ResourcesSourceOutputDirectory>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(TargetGroup)' == 'netcore50aot'">
+  <PropertyGroup Condition="'$(TargetGroup)' == 'netcore50aot'  or '$(TargetGroup)' == 'netstandard13aot'">
     <SkipCommonResourcesIncludes>true</SkipCommonResourcesIncludes>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <ExcludeAssemblyInfoPartialFile>true</ExcludeAssemblyInfoPartialFile>
     <StringResourcesPath>Resources/Strings.netcore50aot.resx</StringResourcesPath>
+    <SkipValidatePackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard13aot'">true</SkipValidatePackageTargetFramework>
```
Comment (ericstj) [inline src/System.IO/src/System.IO.csproj]: Why was this required?

File: src/System.IO/src/System.IO.csproj [hunk 4]
```diff
@@ -76,9 +82,14 @@
...
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)' != 'netstandard13aot'">
     <None Include="project.json" />
   </ItemGroup>
+  
+  <ItemGroup Condition="'$(TargetGroup)' == 'netstandard13aot'">
+    <TargetingPackReference Include="System.Private.Interop" />
+    <None Include="netstandard13aot\project.json" />
```
Comment (ericstj) [inline src/System.IO/src/System.IO.csproj]: Can you add a comment here that says you needed to fork the dependency group for AOT vs non-AOT but could not due to https://github.com/dotnet/cli/issues/465 & https://github.com/nuget/home/issues/1660

File: src/System.IO/src/System.IO.csproj (hunk 3 above)
Comment (ericstj) [inline src/System.IO/src/System.IO.csproj]: If you just needed to exclude the System.Private references you can use ValidateIgnoreReference to do so.

File: src/System.IO/src/System.IO.csproj (hunk 4 above)
Comment (ericstj) [inline src/System.IO/src/System.IO.csproj]: Even better if you don't need to fork it just put this in the main project.json.

**Comment by tarekgh** [comment]
@ericstj fixed the code according to your comments.

**Comment by jasonwilliams200OK** [comment]
You may as well want to run `git commit --amned` and update the commit message to netstandard13aot.

**Comment by tarekgh** [comment]
@jasonwilliams200OK I have updated the commit message. thanks

**Comment by tarekgh** [comment]
@dotnet-bot could you please test this?

**Comment by ericstj** [comment]
Builds are failing because you missed the update to system.io.builds.

File: src/System.IO/src/System.IO.csproj [hunk 5]
```diff
@@ -79,6 +84,11 @@
   <ItemGroup>
     <None Include="project.json" />
   </ItemGroup>
+  
+  <ItemGroup Condition="'$(TargetGroup)' == 'netstandard13aot'">
+    <TargetingPackReference Include="System.Private.Interop" />
+    <ValidateIgnoreReference Include="System.Private.Interop" />
```
Comment (weshaggard) [inline src/System.IO/src/System.IO.csproj]: This can be generalized to `<ValidateIgnoreReference Inlcude="@(TargetingPackReference)" />` I know it is only one in this case but it might be better to generalize this. If we start getting more of these libraries we should actually add this in a common target in BuildTools.

**Comment by ericstj** [comment]
LGTM, OK to merge once all builds pass.

File: src/System.IO/src/System.IO.csproj (hunk 5 above)
Comment (tarekgh) [inline src/System.IO/src/System.IO.csproj]: I updated that. it is a good idea anyway

**Comment by tarekgh** [comment]
thanks @ericstj for your help and review here

**Comment by tarekgh** [comment]
@corivera could you have a look at the test failures here? it looks not related to me changes here:

14:39:53 A :    at System.Data.SqlClient.ManualTesting.Tests.DataTestClass.RunTest() 
14:39:53 E : Error 4 
14:39:53 ** MISSING 
14:39:53 5 : Error 46 : Error 57 : 08 : 19 : 310 : 411 : 512 : 613 : 714 : 815 : 916 : 1017 : 1118 : **** ExecuteScalar ****19 : *** SQL CONNECTION INFO MESSAGE : 0 ****20 : **** ExecuteReader ****21 : *** SQL CONNECTION INFO MESSAGE : 0 ****22 : NextResult23 : *** SQL CONNECTION INFO MESSAGE : 1 ****24 : SqlException: Error 125 : NextResult26 : *** SQL CONNECTION INFO MESSAGE : 3 ****27 : NextResult28 : *** SQL CONNECTION INFO MESSAGE : 4 ****29 : SqlException: Error 230 : NextResult31 : *** SQL CONNECTION INFO MESSAGE : 5 ****32 : NextResult33 : *** SQL CONNECTION INFO MESSAGE : 6 ****34 : SqlException: Error 335 : NextResult36 : *** SQL CONNECTION INFO MESSAGE : 7 ****37 : NextResult38 : *** SQL CONNECTION INFO MESSAGE : 8 ****39 : SqlException: Error 440 : NextResult41 : *** SQL CONNECTION INFO MESSAGE : 9 ****42 : NextResult43 : *** SQL CONNECTION INFO MESSAGE : 10 ****44 : SqlException: Error 545 : NextResult46 : *** SQL CONNECTION INFO MESSAGE : 11 ****
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.DataTestClass.GetConnectionStringFromXml(String key)
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.DataTestClass.GetConnectionString(String connectionStringName)
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.DataTestClass.get_SQL2008_Master()
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.DateTimeTest.TypeVersionKnobTest()
14:39:53    System.Data.SqlClient.ManualTesting.Tests.SqlTypeTest.SqlStringValidComparisonTest [FAIL]
14:39:53       Error (Default Comparison Option with Operator): ﾌｧｽﾞ･ｷﾞﾀｰ and ファズ・ギター should be equal
14:39:53       Expected: True
14:39:53       Actual:   False
14:39:53       Stack Trace:
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.SqlTypeTest.SqlStringDefaultCompareOptionTest(Int32 localeID)
14:39:53             at System.Data.SqlClient.ManualTesting.Tests.SqlTypeTest.SqlStringValidComparisonTest()
14:39:53    System.Data.SqlClient.ManualTesting.Tests.TransactionTest.TestKatmai [FAIL]
14:39:53       System.NullReferenceException : Object reference not set to an instance of an object.
14:39:53       Stack Trace:

**Comment by tarekgh** [comment]
test Innerloop Ubuntu Release Build and Test please

**Comment by tarekgh** [comment]
test Innerloop CentOS7.1 Release Build and Test 
test Innerloop OSX Release Build and Test
test Innerloop Ubuntu Debug Build and Test 
test Innerloop CentOS7.1 Debug Build and Test

**Comment by tarekgh** [comment]
test Innerloop CentOS7.1 Debug Build and Test  please

**Comment by tarekgh** [comment]
test Innerloop CentOS7.1 Release Build and Test please
